plugins/destination/mysql/client/schema.go

```diff
@@ -172,15 +160,15 @@ func (c *Client) createTable(ctx context.Context, table *schema.Table) error {
 	if len(primaryKeysIndices) > 0 {
 		builder.WriteString(",\n  ")
 		builder.WriteString(" PRIMARY KEY (")
+		lengthPerPk := c.maxIndexLength / len(primaryKeysIndices)
```

Comment:
A few things:
1. This makes the assumption that all keys in the composite PK have the same length... So in cases where the PK is something like `req_region`, `req_account_id` and `arn` the first 2 columns will be vastly over provisioned as we know an account_id is always 12 characters.
2. Don't we need to factor in the type of each column in the PK? If one column is an int, doesn't that take up a lot less space than an text/blob that requires the subpart?

Reply:
1. Yes, but we don't have any way of knowing that in advance unless the source adds a column length in bytes to each column. Please note that currently we are still provisioning the same length for each key which is `191`, so this PR doesn't make it worse, only better.

2. Yes it's not 100% yet, I commented on it in the PR description, see https://github.com/cloudquery/cloudquery/pull/13569#issue-1877070038. Please note this probably only impacts tables with many PKs, so this optimization might not be needed yet.